File: POO JAVA/Aula13/src/aula13/Cachorro.java
package aula13;

public class Cachorro extends Lobo{

    //Metodo de Sobreposição da Classe Lobo
    @Override
    public void emitirSom() {
        System.out.println("Au! Au! Au! ...");
    }
    
    //Metodo da Classe
    public void reagir(String frase){
        if(("Comida".equals(frase)) || ("oi".equals(frase))){
            System.out.println("\nAbana o rabinho e latir alegre.");
        }else{
            System.out.println("\nRosna bastante zangado.");
        }
    }
    public void reagir(int hora, int min){
        if(hora < 12){
            System.out.println("\nAbana o rabo ...");
        }else if(hora >= 18){
            System.out.println("\nIgnora a pessoa");
        }else{
            System.out.println("\nAbana e late muito.");
        }
    }
    public void reagir(boolean dono){
        if(dono){
            System.out.println("\nLate muito, alegre ...");
        }else{
            System.out.println("\nRosna e Latir.");
        }
    }
    public void reagir(int idade, float peso){
        if(idade < 2){
            if(peso >1){
                System.out.println("\nCorre, pula e late ...");
            }else{
                System.out.println("\nPula e Corre.");
            }
        }else{
            if(idade >= 2){
                System.out.println("\nLate e se esconde.");
            }else{
                System.out.println("\nEsconde e rosna muito.");
            }
        }
    }
}

File: POO PHP/Aula13/Cachorro.php
<?php

require_once 'Lobo.php';

class Cachorro extends Lobo{

    //Metodo de Sobreposição
    public function emitirSom() {
        echo "<p>Au! Au! Au!!</p>";
    }
    
    //Metodos da classe
    public function reagirFrase($frase) {
        if($frase == "comida" || $frase == "oi"){
            echo "<p>Abanar o rabo e latir.</p>";
        } else {
            echo "<p>Rosnar.</p>";
        }
    }
    public function reagirHoraMin($hora, $min) {
        if($hora < 12){
            echo "<p>Latir e abanar o Rabo.</p>";
        } else if($hora >= 12 && $hora <18){
            echo "<p>Ignorar.</p>";
        } else {
            echo "<p>Latir e correr.</p>";
        }
    }
    public function reagirDono($dono) {
        if($dono){
            echo "<p>Abana o rabinho ...</p>";
        } else {
            echo "<p>Latir muito.</p>";
        }
    }
    public function reagirIdadePeso($idade, $peso) {
        if($idade < 4){
            if($peso < 2){
                echo "<p>Correr.</p>";
            }else{
                echo "<p>Latir.</p>";
            }
        }else {
            if($peso < 2){
                echo "<p>Se esconder.</p>";
            } else {
                echo "<p>Abanar o rabo.</p>";
            }
        }
            
        
    }
}
